### PR TITLE
[12.0][MIG] auth_ldaps

### PR DESCRIPTION
**Pull request by alexey-pelykh**

### CODE REVIEW

**Comment by pedrobaeza** [comment]
Thanks for the contribution!

runbot warnings:
```
2018-10-16 06:26:51,079 150 WARNING openerp_test odoo.models: The model res.company.ldap has no _description
```
Have you signed the CLA?

**Comment by alexey-pelykh** [comment]
re: CLA - yes, just email one.
re: warnings - not sure what to add there as [original model](https://github.com/odoo/odoo/blob/12.0/addons/auth_ldap/models/res_company_ldap.py) does not specify any

**Comment by pedrobaeza** [comment]
Well, you can add the description on your inherited model.

**Comment by alexey-pelykh** [comment]
I'm not sure that adding a description in inherited model is a good approach, since eventually original model may get one. Is there a strict requirement of having that description in inherited model?

**Comment by pedrobaeza** [comment]
This is for avoiding the warning on the runbot, so the CIs are green, so it's mandatory on that way. Now you have also Travis red.

**Comment by alexey-pelykh** [comment]
Yeah, lint :) Fixed both issues, hopefully would work now

**Comment by alexey-pelykh** [comment]
@pedrobaeza I've [added the description](https://github.com/OCA/server-auth/pull/42/files#diff-5b55b2aab03d2c46ea5eeb080d8bbbebR11) yet it still fails for some reason, a hint would be much appreciated

**Comment by alexey-pelykh** [comment]
Strictly speaking, I don't see a `_description` for this case [here](https://www.odoo.com/documentation/12.0/reference/orm.html#extension), please advise

**Comment by pedrobaeza** [comment]
Hi @alexey-pelykh, it turns out that the trick of putting the description in the inherited model doesn't server for avoiding the warning, so here the only solution is to ignore the warning and tell Odoo to fix it in the core (or make a PR fixing it).

**Comment by alexey-pelykh** [comment]
Got it. Are there any other changes needed for this PR to be accepted?

**Comment by pedrobaeza** [comment]
Well, now this one should be technically and functionally reviewed (mine was only "kick-off review"). Check review guidelines in https://github.com/OCA/maintainer-tools/blob/master/CONTRIBUTING.md#review

If you want to speed up your PR, you can review other PRs and ask in exchange that they review yours.

**Comment by alexey-pelykh** [comment]
Thanks for the hint! Just to make sure I've understood you correctly, now the ball is on OCA side and I should just wait?

**Comment by pedrobaeza** [comment]
No, that's not what I mean. It's not on the OCA side, but on contributors side, and also on your side. If you want your PR to be reviewed, you should do the same on other PRs. That's how collaborative efforts work!

File: auth_ldaps/models/res_company_ldap.py
Comment (sebalix) [inline auth_ldaps/models/res_company_ldap.py]: There was an issue regarding this import: https://github.com/OCA/server-auth/issues/37
Can you fix it by protecting the import?
```python
try:
    import ldap
except (ImportError) as err:
    _logger.debug(err)
```
https://github.com/OCA/maintainer-tools/blob/master/CONTRIBUTING.md#importerror

Comment (sebalix) [reply auth_ldaps/models/res_company_ldap.py]: And put it in a separate commit (no squash) to backport it easily on 11.0.

File: auth_ldaps/__manifest__.py
Comment (lmignon) [inline auth_ldaps/__manifest__.py:18]: The external dependency on ldap is missing
```python
'external_dependencies': {
        'python': [
            'ldap',
        ],
    },
```

And put it in a separate commit (no squash) to backport it easily on 11.0. :smirk:

**Review by lmignon — CHANGES_REQUESTED** [review]
Thank you @alexey-pelykh fot this work.
A little comment. (Code review only)

File: auth_ldaps/__manifest__.py
Comment (alexey-pelykh) [reply auth_ldaps/__manifest__.py:18]: This should be on `auth_ldap` module in Odoo itself I guess, since this module expects that one to be present.

File: auth_ldaps/models/res_company_ldap.py
Comment (alexey-pelykh) [reply auth_ldaps/models/res_company_ldap.py]: Even if it's a mandatory dependency of the "parent" module?

File: auth_ldaps/__manifest__.py
Comment (lmignon) [reply auth_ldaps/__manifest__.py:18]: It should not be assumed that dependency is resolved by transitivity. The best practice is to declare the dependency everywhere it's used.  In your module you explicitly import the ldap module. Therefore you have a direct dependency on this module and it should be declared into the manifest.

File: auth_ldaps/models/res_company_ldap.py
Comment (alexey-pelykh) [reply auth_ldaps/models/res_company_ldap.py]: Done

File: auth_ldaps/__manifest__.py
Comment (alexey-pelykh) [reply auth_ldaps/__manifest__.py:18]: Done

Comment (lmignon) [reply auth_ldaps/__manifest__.py:18]: Thank you @alexey-pelykh

**Review by lmignon — APPROVED** [review]
LGTM **Code review only**

**Review by sebalix — APPROVED** [review]

**Comment by alexey-pelykh** [comment]
Next step on my side is to wait, right?

**Comment by alexey-pelykh** [comment]
Runbot warnings are going to be present until https://github.com/odoo/odoo/pull/28047 is merged

**Comment by alexey-pelykh** [comment]
@pedrobaeza all green and 5 days seems to have passed, what should I do next?

**Comment by pedrobaeza** [comment]
Please squash your 3 migration commits in one following https://github.com/OCA/maintainer-tools/wiki/Merge-commits-in-pull-requests and I'll merge.

**Comment by alexey-pelykh** [comment]
But there was a specific request to make those 2 changes separately by @lmignon and @sebalix to backport those to 10.0 and 11.0

**Comment by pedrobaeza** [comment]
@lmignon and I have different notion of commits. I'm more commit saving as they don't usually express so much being split (although there are of course exceptions), but he seems to collect commits :yum: 

I think you should at least squash last 2 as they are about the same thing and will be cherry-picked together, but I let you the final word.

**Comment by alexey-pelykh** [comment]
Sounds reasonable! Done

**Comment by lmignon** [comment]
@pedrobaeza In this specific case merging the two last commit makes senss

**Comment by alexey-pelykh** [comment]
@pedrobaeza do these modules get released to pypi.org?

**Comment by pedrobaeza** [comment]
It should through our bots, but I'm not too aware of the process. I'm seeing the setup directory in https://github.com/OCA/server-auth/tree/11.0/setup/auth_ldaps, but I can't say if everything is OK. @lmignon or @sbidoul can say about that.

**Comment by sbidoul** [comment]
@alexey-pelykh It is on pypi: https://pypi.org/project/odoo12-addon-auth-ldaps/#history

**Comment by alexey-pelykh** [comment]
Thanks! For some reason, search did not bring it up

**Comment by alexey-pelykh** [comment]
And to https://apps.odoo.com/apps as well?

**Comment by sbidoul** [comment]
Ah that's another story. Someone has to manually add the 200 or so 12.0 branches on apps.odoo.com... I don't think that has been done yet.

**Pull request merged**
